sdk/Lusid.Sdk.Tests/tutorials/Instruments/DemoEquityOption.cs

```diff
@@ -0,0 +1,181 @@
+            var equityOption = (EquityOption) InstrumentExamples.CreateExampleEquityOption();
+            Assert.That(equityOption, Is.Not.Null);
+
+            // Can now UPSERT to Lusid
+            string uniqueId = "id-equityOption-1";
+            _testDataUtilities.UpsertOtcToLusid(equityOption, "some-name-for-this-equityOption", uniqueId);
+
+            // Can now QUERY from Lusid
+            var retrieved = _testDataUtilities.QueryOtcFromLusid(uniqueId);
+            Assert.That(retrieved.InstrumentType == LusidInstrument.InstrumentTypeEnum.EquityOption);
+            var roundTripEquityOption = retrieved as EquityOption;
+            Assert.That(roundTripEquityOption, Is.Not.Null);
+            Assert.That(roundTripEquityOption.Code, Is.EqualTo(equityOption.Code));
+            Assert.That(roundTripEquityOption.Strike, Is.EqualTo(equityOption.Strike));
+            Assert.That(roundTripEquityOption.DeliveryType, Is.EqualTo(equityOption.DeliveryType));
+            Assert.That(roundTripEquityOption.DomCcy, Is.EqualTo(equityOption.DomCcy));
+            Assert.That(roundTripEquityOption.OptionType, Is.EqualTo(equityOption.OptionType));
+            Assert.That(roundTripEquityOption.StartDate, Is.EqualTo(equityOption.StartDate));
+            Assert.That(roundTripEquityOption.OptionMaturityDate, Is.EqualTo(equityOption.OptionMaturityDate));
+            Assert.That(roundTripEquityOption.OptionSettlementDate, Is.EqualTo(equityOption.OptionSettlementDate));
+            Assert.That(roundTripEquityOption.UnderlyingIdentifier, Is.EqualTo(equityOption.UnderlyingIdentifier));
+        }
+
+        [Test]
+        public void DemoEquityOptionValuation()
+        {
+            var scope = "DemoEquityOptionValuation";
+
+            // upsert option
+            var option = InstrumentExamples.CreateExampleEquityOption();
+            string uniqueId = "id-equityOption-1";
+            _testDataUtilities.UpsertOtcToLusid(option, "some-name-for-this-equityOption", uniqueId);
+
+            // for Black-Scholes pricing, we need the following market data
+            _testDataUtilities.CreateAndUpsertSimpleQuote(scope, "ACME", QuoteSeriesId.InstrumentIdTypeEnum.RIC, 110m, "USD", _demoEffectiveAt);
+            _testDataUtilities.CreateAndUpsertOisCurve(scope, _demoEffectiveAt, "USD");
+            _testDataUtilities.CreateAndUpsertVolSurface(scope, _demoEffectiveAt, option, 0.2m);
+
+            // create Black-Scholes recipe specifying where to look for market data and which metrics to return
+            // if in a larger portfolio, we would make a specific VendorModelRule specifying that equity options are to be valued using Black-Scholes
+            var recipeCode = "EquityOption_ValuationRecipe";
+            var pricingOptions = new PricingOptions(new ModelSelection(ModelSelection.LibraryEnum.Lusid, ModelSelection.ModelEnum.BlackScholes));
+            var recipe = new ConfigurationRecipe
```

Comment:
Think it'd be a nice touch to
```
[TestCase(CTVoM)]
[TestCase(DCC)]
[TestCase(BS)]
```
as it shows us what models we have done and also how easy it is to switch between models.

Reply:
I would agree, but in this case they all give the same output so don't see the point for now. Happy to do it if you wish, however I would put the relevant depenency upserts in their own "if" blocks, as I like each test to only be upserting the minimal amount of market data necessary.

Reply:
Same output? The numbers should in principal change (granted we do not assert on that). 
I think it'd be good to do. Changing model in other systems can take N days, the fact we can by changing one field is a one of the (mini?) USPs. You can decide how you want to upsert data in minimal (or not) way.

Reply:
The numbers don't change (as you said, physically settled options assume exercise), and discounting doesn't discount (which I would argue is fine)

Reply:
Ah right, this is for cashflows. Thought it was GetValuation. Can we vary the models for GetValuation?

For cashflow call: it's still nice to show we get the cashflow across the models? For our sanity, it shows it works across the models also (different models can run through code slightly differently)

Reply:
Can even put in Bachelier too. CTVoM and B definitely runs slightly different path to DCC/BS (but should all join back up)